### PR TITLE
[11.x] PHPUnit 11 Forward Compatibility

### PR DESCRIPTION
**Pull request by crynobone**
* [x] Remove PHPUnit docblock annotation and replace with PHP Attributes.

The following has been done in `10.x` via #48815 :
* [x] Replace `getMockForTrait()` https://github.com/sebastianbergmann/phpunit/issues/5306
* [x] Replace `getMockForAbstractClass()` https://github.com/sebastianbergmann/phpunit/issues/5305
* [x] Replace `addMethods()` https://github.com/sebastianbergmann/phpunit/issues/5320

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -107,7 +107,7 @@
         "orchestra/testbench-core": "^9.0",
         "pda/pheanstalk": "^4.0",
         "phpstan/phpstan": "^1.4.7",
-        "phpunit/phpunit": "^10.1",
+        "phpunit/phpunit": "^11.0",
```
Comment (crynobone) [inline composer.json:110]:
```suggestion
        "phpunit/phpunit": "^10.1|^11.0",
```

**Comment by GrahamCampbell** [comment]
I would prefer to see just supporting PHPUnit 10 on Laravel 11, and PHPUnit 11 on Laravel 12, etc, giving us the much more stable and final versions of PHPUnit in each case.

**Comment by crynobone** [comment]
Everything in this PR still will work fine with PHPUnit 10.x, we can't/shouldn't send annotation to attribute changes to because it still supports both PHPUnit 9.x and 10.x

**Review by driesvints — APPROVED** [review]

**Comment by GrahamCampbell** [comment]
OK, this is better now that it doesn't actually use PHPUnit 11. ;)

**Pull request merged**